app/screens/edit_profile/edit_profile.tsx

```diff
@@ -120,19 +137,110 @@ const EditProfile = ({
         setCanSave(value);
     }, [componentId, rightButton, isTablet]);
 
+    // Get database and observables
+    const database = DatabaseManager.getServerDatabaseAndOperator(serverUrl).database;
+    const dbAttributes = useState(() =>
+        (enableCustomAttributes ? observeCustomProfileAttributesByUserId(database, currentUser?.id || '') : null),
+    )[0];
+    const dbFields = useState(() =>
+        (enableCustomAttributes ? observeCustomProfileFields(database) : null),
+    )[0];
+
     useEffect(() => {
         const loadCustomAttributes = async () => {
             if (!currentUser) {
                 return;
             }
-            const {error: fetchError, attributes} = await fetchCustomAttributes(serverUrl, currentUser.id);
-            if (!fetchError && attributes) {
-                setUserInfo((prev: UserInfo) => ({...prev, customAttributes: attributes} as UserInfo));
+
+            try {
+                // First load from database
+                if (enableCustomAttributes) {
+                    const attrs = await queryCustomProfileAttributesByUserId(database, currentUser.id).fetch();
+                    if (attrs.length) {
+                        const converted = await convertProfileAttributesToCustomAttributes(
+                            database,
+                            attrs,
+                            sortCustomProfileAttributes,
+                        );
+                        if (converted.length) {
+                            setUserInfo((prev) => ({
+                                ...prev,
+                                customAttributes: converted.reduce(customAttributesReducer, {} as Record<string, CustomAttribute>),
+                            }));
+                        }
+                    }
+                }
+            } catch (dbError) {
+                logError('Error loading custom attributes from database', dbError);
+            }
```

Comment:
why are we doing this here, this should be done in the `withObservables` HOC and you can just observe the first output and not listen to the rest if that is what you want to achieve.